daemon/emer-cache-size-provider.c

```diff
@@ -44,199 +34,55 @@ G_DEFINE_TYPE_WITH_PRIVATE (EmerCacheSizeProvider, emer_cache_size_provider, G_T
-{
-  return g_object_new (EMER_TYPE_CACHE_SIZE_PROVIDER, NULL);
-}
-
-/*
- * emer_cache_size_provider_new_full:
- * @path: path to a file specifying a maximum persistent cache size; see
- * #EmerCacheSizeProvider:path.
- *
- * Constructs a provider that obtains the maximum persistent cache size from the
- * given filepath.
+ * emer_cache_size_provider_get_max_cache_size:
+ * @path: (allow-none): the path to the file where the maximum persistent
+ *  cache size is stored.
  *
- * Returns: (transfer full): A new #EmerCacheSizeProvider.
- * Free with g_object_unref().
+ * Returns the maximum persistent cache size in bytes. If @path is %NULL, it
+ * defaults to DEFAULT_CACHE_SIZE_FILE_PATH. If the underlying configuration
+ * file doesn't exist yet, or is corrupt, it is recreated with a default value
+ * of DEFAULT_MAX_CACHE_SIZE.
  */
-EmerCacheSizeProvider *
-emer_cache_size_provider_new_full (const gchar *path)
-{
-  return g_object_new (EMER_TYPE_CACHE_SIZE_PROVIDER,
-                       "path", path,
-                       NULL);
-}
-
-static void
-write_cache_size (EmerCacheSizeProvider *self,
-                  guint64                max_cache_size)
+guint64
+emer_cache_size_provider_get_max_cache_size (const gchar *path)
 {
-  EmerCacheSizeProviderPrivate *priv =
-    emer_cache_size_provider_get_instance_private (self);
+  g_autoptr(GKeyFile) key_file = g_key_file_new ();
+  guint64 cache_size;
+  gboolean load_succeeded;
+  g_autoptr(GError) error = NULL;
 
-  g_key_file_set_uint64 (priv->key_file, CACHE_SIZE_GROUP,
-                         MAX_CACHE_SIZE_KEY, max_cache_size);
+  if (path == NULL)
+    path = DEFAULT_CACHE_SIZE_FILE_PATH;
 
-  GError *error = NULL;
-  gboolean save_succeeded =
-    g_key_file_save_to_file (priv->key_file, priv->path, &error);
-  if (!save_succeeded)
+  if (g_key_file_load_from_file (key_file, path, G_KEY_FILE_NONE,
+                                  &error))
     {
-      g_warning ("Failed to write default cache size file to %s. Error: %s.",
-                 priv->path, error->message);
-      g_error_free (error);
+      cache_size = g_key_file_get_uint64 (key_file, CACHE_SIZE_GROUP,
+                                          MAX_CACHE_SIZE_KEY, &error);
     }
-}
-
-static void
-read_cache_size_data (EmerCacheSizeProvider *self)
-{
-  EmerCacheSizeProviderPrivate *priv =
-    emer_cache_size_provider_get_instance_private (self);
 
-  if (priv->data_cached)
-    return;
-
-  GError *error = NULL;
-  gboolean load_succeeded =
-    g_key_file_load_from_file (priv->key_file, priv->path, G_KEY_FILE_NONE,
-                               &error);
-  if (!load_succeeded)
+  if (error != NULL)
     {
+      /* If the file doesn't exist, silently create it. Otherwise, something
+       * was wrong with it:
+       */
       if (!g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
-        goto handle_failed_read;
-
+        {
+          g_warning ("Error reading cache size from %s: %s", path,
```

Comment:
Looking again, I think we should ignore `G_KEY_FILE_ERROR_GROUP_NOT_FOUND` and `G_KEY_FILE_ERROR_KEY_NOT_FOUND` and just silently add the key.